scripts/asc_autosubmit.rb

```
@@ -34,30 +34,20 @@
-  return [res&.code.to_i, parsed]
-  warn "⚠️ Failed to parse JSON response for #{method.to_s.upcase} #{path} (status=#{status}): #{e.message}; body=#{body_preview.inspect}"
+  parsed = JSON.parse(res.body)
 rescue JSON::ParserError => e
-  status = res&.code || 'no-response'
-  return [(res&.code || 0).to_i, parsed]
-  warn "⚠️ Failed to parse JSON response for #{method.to_s.upcase} #{path} (status=#{status}): #{e.message}; body=#{body_preview}"
+  warn "⚠️ Failed to parse JSON response for #{method.to_s.upcase} #{path} (status=#{res.code}): #{e.message}; body=#{res.body.to_s[0, 500].inspect}"
   parsed = {}
 ensure
-  return [(res&.code || 0).to_i, parsed]
+  return [res.code.to_i, parsed]
```

RuboCop (1.88.0)

[warning] Do not return from an `ensure` block.

(Lint/EnsureReturn)